### ISSUE TITLE
IPv6 examples using invalid internal addresses

### ISSUE THREAD

**Issue by timriker** [Azure/azure-quickstart-templates#7158]
### Template files

- ipv6-in-vnet/azuredeploy.json
- ipv6-in-vnet-StdLB/azuredeploy.json
- ipv6-in-vnet-vmss/azuredeploy.json

### Issue Details
These address use

`ace:cab:deca::/48`

ipv6 addresses for internal networks which appear to be in the reserved address block.

I believe Internal networks should be using the Unique Local address space of:

`fc00::/7`

instead. All templates and training materials should be updated to avoid using reserved address space. As the internal networks are not routable, this address space seems to be the best choice.

`fcce:cab:deca::/48` would work fine. Not sure where the current invalid ipv6 space came from. It would be better to to generate a unique Global ID which might then allow different groups to route traffic eventually.

https://tools.ietf.org/html/rfc4193
https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml

**Comment by denisbrodbeck** [comment]
I agree, Admins/Devs should be made aware, that they should use a ULA (Unique Local Address) space, when assigning ipv6 address space to their virtual networks. Emphasis on **unique**.

Here's what [RFC 4193](https://tools.ietf.org/html/rfc4193) says:

> 3.2.1.  Locally Assigned Global IDs
>
> Locally assigned Global IDs MUST be generated with a pseudo-random
> algorithm consistent with [RANDOM].  Section 3.2.2 describes a
> suggested algorithm.  It is important that all sites generating
> Global IDs use a functionally similar algorithm to ensure there is a
> high probability of uniqueness.
>
> The use of a pseudo-random algorithm to generate Global IDs in the
> locally assigned prefix gives an assurance that any network numbered
> using such a prefix is highly unlikely to have that address space
> clash with any other network that has another locally assigned prefix
> allocated to it.  This is a particularly useful property when
> considering a number of scenarios including networks that merge,
> overlapping VPN address space, or hosts mobile between such networks.

This tool creates random ULAs: https://cd34.com/rfc4193/

**Comment by jlaundry** [comment]
FYI, as of November 2020 the Azure team had this on a backlog somewhere: https://github.com/MicrosoftDocs/azure-docs/pull/46099
